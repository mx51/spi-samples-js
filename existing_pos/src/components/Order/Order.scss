@import '../ColorPalette';

div.order-sidebar {
  border-left: 1px solid $secondaryGrey;
  padding: 0;
  min-height: 100vh;
}

.removeStyle {
  list-style-type: none;
  padding: 1rem;
}

.total {
  position: absolute;
  bottom: 4rem;
  right: 0;
  left: 0;
  padding: 1rem;
  color: $primaryGrey;
  font-size: 1rem;
  letter-spacing: 1.25px;
}

.quantity-buttons > button {
  display: inline-block;
  width: 2rem;
}

.quantity-buttons > div.quantity-label {
  border: 1px solid $tertiary;
  display: inline-block;
  padding: 0.25rem 0.7rem 0.5rem 0.8rem;
}
.quantity {
  padding-top: 7px;
}

.order-header-buttons button {
  width: 100%;
  height: 3rem;
  background: none;
  font-size: 0.9rem;
}

button.checkout-button {
  position: absolute;
  bottom: 0;
  height: 4.5rem;
  font-size: 1.5rem;
}

.orderListScroll {
  height: calc(100vh - 20rem);
  overflow-y: scroll;
  padding: 0;
  margin: 0;

  li::before {
    content: "";
    display: inline-block;
    width: 1.25rem;
  }
}
