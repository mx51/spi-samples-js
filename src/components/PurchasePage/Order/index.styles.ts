import { makeStyles, Theme, createStyles, alpha } from '@material-ui/core/styles';

export default makeStyles((theme: Theme) =>
  createStyles({
    root: {
      height: `calc(100vh - 48px)`,
      position: 'sticky',
      top: theme.spacing(6),
      zIndex: 1020,
    },
    heading: {
      padding: theme.spacing(2),
    },
    clear: {
      color: theme.palette.info.main,
      fontSize: '1rem',
      lineHeight: '1.5rem',
      fontWeight: 'lighter',
    },
    keypadDrawerPaper: {
      width: '33.34%',
      backgroundColor: 'transparent',
      border: 0,
    },
    total: {
      color: theme.palette.primary.main,

      fontWeight: 500,
      fontSize: '1.25rem',
      lineHeight: '1.5rem',
      letterSpacing: '0.03rem',
    },

    totalPrice: {
      fontSize: '2rem',
      color: theme.palette.primary.main,
    },
    payNowBtn: {
      padding: theme.spacing(2),
      borderRadius: 0,
      backgroundColor: theme.palette.error.dark,
      '&:hover, &:active, &:focus': {
        background: alpha(theme.palette.error.dark, 0.75),
        color: theme.palette.common.white,
      },
    },
    amendBtn: {
      padding: theme.spacing(2),
      borderRadius: 0,
      backgroundColor: theme.palette.text.hint,
      '&:hover, &:active, &:focus': {
        background: alpha(theme.palette.text.hint, 0.75),
        color: theme.palette.common.white,
      },
      fontWeight: 100,
    },
    actionBtnLabel: {
      fontWeight: 400,
      fontSize: '1.5rem',
      lineHeight: '1.5rem',
    },
    items: {
      color: theme.palette.primary.main,
      fontSize: '1rem',
      fontWeight: 'normal',
      lineHeight: '1.5rem',
    },
    amount: {
      fontSize: '1.25rem',
    },
    orderList: {
      flexGrow: 1,
      overflowY: 'auto',
    },
    checkbox: {
      paddingLeft: theme.spacing(3),
      color: theme.palette.primary.main,
    },
    checkboxLabel: {
      fontSize: '0.875rem',
    },
  })
);
