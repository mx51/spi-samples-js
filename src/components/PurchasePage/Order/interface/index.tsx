export interface IOrderProps {
  disablePayNow: boolean;
}
