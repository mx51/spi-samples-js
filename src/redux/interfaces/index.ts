import { IProductState } from '../reducers/ProductSlice/interfaces';
import { ITerminalState } from '../reducers/TerminalSlice/interfaces';

export interface ISamplePosState {
  terminals: ITerminalState;
  products: IProductState;
}
