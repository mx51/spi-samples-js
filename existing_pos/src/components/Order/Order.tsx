import React, { useState } from 'react';
import { SpiStatus } from '@mx51/spi-client-js';
import { Col, Row, Modal, Button } from 'react-bootstrap';
import GetTransactionModal from '../GetTransactionModal';
import SurchargeModal from '../SurchargeModal';
import './Order.scss';

function removeProductQuantityAction(
  id: string,
  list: Array<Product>,
  handleApplySurcharge: Function,
  onChangeProductQuantity: Function
) {
  if (list.length === 1 && list[0].quantity === 1) {
    handleApplySurcharge(0);
  }
  onChangeProductQuantity(id, -1);
}

function checkoutAction(status: String, onErrorMsg: Function, onCheckout: Function) {
  if (status !== SpiStatus.PairedConnected) {
    onErrorMsg('Please pair your POS to the terminal or check your network connection');
  } else {
    onCheckout();
  }
}
function Order(props: {
  list: Array<Product>;
  onCheckout: Function;
  onGetTransaction: Function;
  onRefund: Function;
  onLastTransaction: Function;
  onChangeProductQuantity: Function;
  handleApplySurcharge: Function;
  posRefId: string;
  setPosRefId: Function;
  surchargeAmount: number;
  status: string;
  errorMsg: string;
  onErrorMsg: Function;
}) {
  const {
    list,
    onCheckout,
    onGetTransaction,
    onRefund,
    onLastTransaction,
    onChangeProductQuantity,
    handleApplySurcharge,
    posRefId,
    setPosRefId,
    surchargeAmount,
    status,
    errorMsg,
    onErrorMsg,
  } = props;

  const [showGetTransactionModal, setShowGetTransactionModal] = useState(false);

  const [showSurcharge, setShowSurcharge] = useState(false);
  const [surcharge, setSurcharge] = useState<number>(0);

  const subTotalAmount: number = list.reduce(
    (total: number, product: Product) => total + parseFloat(product.price) * product.quantity,
    0
  );
  const totalAmount = subTotalAmount + surchargeAmount / 100;

  return (
    <div className="min-vh-100 sticky-top">
      <GetTransactionModal
        posRefId={posRefId}
        setPosRefId={setPosRefId}
        show={showGetTransactionModal}
        handleClose={() => setShowGetTransactionModal(false)}
        handleGetTransaction={() => {
          onGetTransaction(posRefId);
          setShowGetTransactionModal(false);
        }}
      />

      <SurchargeModal
        show={showSurcharge}
        surcharge={surcharge}
        setSurcharge={setSurcharge}
        handleClose={() => setShowSurcharge(false)}
        handleApplySurcharge={() => {
          handleApplySurcharge(surcharge);
          setShowSurcharge(false);
        }}
      />

      <h2 className="sub-header mb-0">Order</h2>
      <Row className="order-header-buttons no-gutters">
        {new URLSearchParams(window.location.search).get('qamode') === 'true' && (
          <Col sm={12}>
            <button
              className="btn btn-outline-primary rounded-0"
              id="lastTransactionButton"
              type="button"
              onClick={() => onLastTransaction()}
            >
              Last Transaction
            </button>
          </Col>
        )}
        <Col sm={4}>
          <button
            className="btn btn-outline-primary rounded-0"
            id="getTransactionButton"
            type="button"
            onClick={() => setShowGetTransactionModal(true)}
          >
            Get Transaction
          </button>
        </Col>
        <Col sm={4}>
          <button
            className="btn btn-outline-primary rounded-0"
            id="surchargeButton"
            type="button"
            onClick={() => setShowSurcharge(true)}
          >
            Add Surcharge
          </button>
        </Col>
        <Col sm={4}>
          <button
            className="btn btn-outline-primary rounded-0"
            id="refundButton"
            type="button"
            onClick={() => onRefund()}
          >
            Refund
          </button>
        </Col>
      </Row>
      <Modal show={errorMsg !== ''} onHide={() => onErrorMsg('')}>
        <Modal.Header closeButton>
          <Modal.Title>Alert</Modal.Title>
        </Modal.Header>
        <Modal.Body>
          <p>{errorMsg}</p>
          <Button variant="primary" className="btn-custom" onClick={() => onErrorMsg('')} block>
            OK
          </Button>
        </Modal.Body>
      </Modal>
      <div className="orderListScroll">
        <ul className="nobull removeStyle">
          {list.map((item: Product) => (
            <li className="space" key={`order-list-${item.id}`}>
              <Row>
                <Col sm={12}>
                  <Row>
                    <Col sm={5}>
                      <div className="quantity-buttons">
                        <button
                          className="btn btn-secondary rounded-0 btn-block"
                          id={`btnItemDec${item.id}`}
                          type="button"
                          onClick={() =>
                            removeProductQuantityAction(item.id, list, handleApplySurcharge, onChangeProductQuantity)
                          }
                        >
                          -
                        </button>
                        <div id={`itemQuantity${item.id}`} className="quantity-label">
                          {item.quantity}
                        </div>
                        <button
                          className="btn btn-secondary rounded-0 btn-block"
                          id={`btnItemInc${item.id}`}
                          type="button"
                          onClick={() => onChangeProductQuantity(item.id, 1)}
                        >
                          +
                        </button>
                      </div>
                    </Col>
                    <Col sm={5} className="quantity">
                      {item.name}
                    </Col>
                    <Col sm={2} className="quantity text-right">
                      ${parseFloat(item.price) * item.quantity}
                    </Col>
                  </Row>
                </Col>
              </Row>
            </li>
          ))}
        </ul>
      </div>

      <div className="total">
        <div>
          <Row>
            <Col sm={7}>Subtotal</Col>
            <Col sm={5} className="text-right">
              ${subTotalAmount.toFixed(2)}
            </Col>
          </Row>
          <Row>
            <Col sm={7}>Surcharge</Col>
            <Col sm={5} className="text-right">
              ${(surchargeAmount / 100).toFixed(2)}
            </Col>
          </Row>
        </div>
        <hr />
        <div className="total-amount">
          <Row>
            <Col sm={7}>Total</Col>
            <Col sm={5} className="text-right">
              ${totalAmount.toFixed(2)}
            </Col>
          </Row>
        </div>
      </div>
      <button
        type="button"
        className="primary-button checkout-button mb-0"
        onClick={() => checkoutAction(status, onErrorMsg, onCheckout)}
      >
        Checkout
      </button>
    </div>
  );
}

export default Order;
