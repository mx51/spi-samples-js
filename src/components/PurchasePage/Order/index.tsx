import React, { useState } from 'react';
import { ListItemText, Box, Button, Divider, List, ListItem, Paper, Typography, Drawer } from '@material-ui/core';
import { useDispatch, useSelector } from 'react-redux';
import { Link as LinkRouter } from 'react-router-dom';
import { PATH_PAY_NOW, PATH_PURCHASE } from '../../../definitions/constants/routerConfigs';
import { IProductSelector } from '../../../redux/reducers/ProductSlice/interfaces';
import {
  orderCashoutAmountSelector,
  orderSurchargeAmountSelector,
  orderTipAmountSelector,
  orderTotalSelector,
  productsSelector,
  productSubTotalSelector,
} from '../../../redux/reducers/ProductSlice/productSelector';
import {
  addCashoutAmount,
  addSurchargeAmount,
  addTipAmount,
  clearAllProducts,
} from '../../../redux/reducers/ProductSlice/productSlice';
import currencyFormat from '../../../utils/common/intl/currencyFormatter';
import KeyPad from '../../KeyPad';
import OrderLineItem from '../../OrderLineItem';
import OrderSubTotal from '../../OrderSubTotal';
import useStyles from './index.styles';
import { IOrderProps } from './interface';

function Order({ disablePayNow }: IOrderProps): React.ReactElement {
  const SURCHARGE_AMOUNT = 'SURCHARGE_AMOUNT';
  const CASHOUT_AMOUNT = 'CASHOUT_AMOUNT';
  const TIP_AMOUNT = 'TIP_AMOUNT';

  const dispatch = useDispatch();

  const products: Array<IProductSelector> = useSelector(productsSelector);
  const subtotalAmount: number = useSelector(productSubTotalSelector);

  const surchargeAmount: number = useSelector(orderSurchargeAmountSelector);
  const cashoutAmount: number = useSelector(orderCashoutAmountSelector);
  const tipAmount: number = useSelector(orderTipAmountSelector);

  const [keypadType, setKeypadType] = useState<string>('');
  const totalAmount: number = useSelector(orderTotalSelector);

  const clearAllProductsAction = () => {
    dispatch(clearAllProducts());
  };

  const requestAmount = (val: string) => {
    setKeypadType(val);
  };

  const getDefaultKeypadAmount = (): number => {
    if (keypadType === SURCHARGE_AMOUNT) return surchargeAmount;
    if (keypadType === CASHOUT_AMOUNT) return cashoutAmount;
    if (keypadType === TIP_AMOUNT) return tipAmount;
    return 0;
  };

  const setKeypadAmount = (val: number) => {
    if (keypadType === SURCHARGE_AMOUNT) dispatch(addSurchargeAmount(val));
    else if (keypadType === CASHOUT_AMOUNT) dispatch(addCashoutAmount(val));
    else if (keypadType === TIP_AMOUNT) dispatch(addTipAmount(val));
    setKeypadType('');
  };

  const getTitle = () => {
    if (keypadType === SURCHARGE_AMOUNT) return 'Surcharge';
    if (keypadType === CASHOUT_AMOUNT) return 'Cashout';
    if (keypadType === TIP_AMOUNT) return 'Tip';
    return '';
  };

  const getSubtitle = () => `Enter ${getTitle().toLowerCase()} amount`;

  const classes = useStyles();
  return (
    <>
      <Drawer
        anchor="right"
        open={keypadType !== ''}
        classes={{
          paper: classes.keypadDrawerPaper,
        }}
      >
        <KeyPad
          open={keypadType !== ''}
          title={getTitle()}
          subtitle={getSubtitle()}
          defaultAmount={getDefaultKeypadAmount()}
          onClose={() => {
            setKeypadType('');
          }}
          onAmountChange={setKeypadAmount}
        />
      </Drawer>
      <Box component={Paper} className={classes.root} display="flex" flexDirection="column">
        <Box display="flex" justifyContent="space-between" className={classes.heading} alignItems="center">
          <Typography variant="h6" component="h1">
            Order
          </Typography>
          <Box>
            {!disablePayNow && (
              <Button className={classes.clear} onClick={clearAllProductsAction}>
                Clear all
              </Button>
            )}
          </Box>
        </Box>
        <Divider />
        <Box className={classes.orderList}>
          <List>
            {products.map((product) => (
              <ListItem key={`product_${product.product.id}`}>
                <ListItemText
                  primary={`${product.quantity} x ${product.product.name}`}
                  classes={{ primary: classes.items }}
                />
                <Typography className={classes.amount}>
                  {currencyFormat((product.product.price * product.quantity) / 100)}
                </Typography>
              </ListItem>
            ))}
          </List>
        </Box>
        <Divider />
        <List>
          <OrderSubTotal label="Subtotal" amount={subtotalAmount} />
          <OrderLineItem
            disabled={false}
            label="Surcharge"
            amount={surchargeAmount}
            onAdd={() => requestAmount(SURCHARGE_AMOUNT)}
            viewOnly={disablePayNow}
          />
          <OrderLineItem
            disabled={tipAmount > 0}
            label="Cashout"
            amount={cashoutAmount}
            onAdd={() => requestAmount(CASHOUT_AMOUNT)}
            viewOnly={disablePayNow}
          />
          <OrderLineItem
            disabled={cashoutAmount > 0}
            label="Tip"
            amount={tipAmount}
            onAdd={() => requestAmount(TIP_AMOUNT)}
            viewOnly={disablePayNow}
          />
          <Divider variant="middle" />
          <ListItem>
            <ListItemText primary="Total" classes={{ primary: classes.total }} />
            <Typography className={classes.totalPrice}>{currencyFormat(totalAmount / 100)}</Typography>
          </ListItem>
        </List>
        {!disablePayNow ? (
          <Button
            variant="contained"
            color="primary"
            size="large"
            classes={{ root: classes.payNowBtn, label: classes.actionBtnLabel }}
            component={LinkRouter}
            to={PATH_PAY_NOW}
          >
            Pay now
          </Button>
        ) : (
          <Button
            variant="contained"
            color="primary"
            size="large"
            classes={{ root: classes.amendBtn, label: classes.actionBtnLabel }}
            component={LinkRouter}
            to={PATH_PURCHASE}
          >
            &lt; Amend order
          </Button>
        )}
      </Box>
    </>
  );
}

export default Order;
