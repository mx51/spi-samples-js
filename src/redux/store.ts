import { AnyAction, Reducer, combineReducers, configureStore } from '@reduxjs/toolkit';
import { persistReducer, persistStore } from 'redux-persist';
import storage from 'redux-persist/lib/storage';
import hardSet from 'redux-persist/es/stateReconciler/hardSet';
import commonReducer from './reducers/CommonSlice/commonSlice';
import pairFormReducer from './reducers/PairFormSlice/pairFormSlice';
import { preAuthReducer } from './reducers/PreAuth/preAuthSlice';
import productSlice from './reducers/ProductSlice/productSlice';
import selectedTerminalReducer from './reducers/SelectedTerminalSlice/selectedTerminalSlice';
import terminalReducer from './reducers/TerminalSlice/terminalsSlice';
import { IPreAuthState } from './reducers/PreAuth/interfaces';
import { PayAtTableState, payAtTableReducer } from './reducers/PayAtTableSlice/payAtTableSlice';
import { ITerminalState } from './reducers/TerminalSlice/interfaces';

const rootReducer = combineReducers({
  common: commonReducer,
  pairForm: pairFormReducer,
  products: productSlice,
  terminals: persistReducer<ITerminalState, AnyAction>(
    {
      key: 'terminals',
      storage,
      stateReconciler: hardSet,
    },
    terminalReducer
  ) as unknown as Reducer<ITerminalState>,
  selectedTerminal: selectedTerminalReducer,
  preAuth: persistReducer<IPreAuthState, AnyAction>(
    {
      key: 'root',
      storage,
      stateReconciler: hardSet,
    },
    preAuthReducer
  ) as unknown as Reducer<IPreAuthState>,
  payAtTable: persistReducer<PayAtTableState, AnyAction>(
    {
      key: 'payAtTable',
      storage,
      stateReconciler: hardSet,
    },
    payAtTableReducer
  ) as unknown as Reducer<PayAtTableState>,
});

export const store = configureStore({
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware({
      serializableCheck: false,
    }),
  reducer: rootReducer,
  devTools: process.env.NODE_ENV === 'development',
});

export const persistor = persistStore(store);

export type RootState = ReturnType<typeof rootReducer>;
export type AppStore = typeof store;
export type AppDispatch = typeof store.dispatch;
