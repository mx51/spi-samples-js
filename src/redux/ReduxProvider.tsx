import React from 'react';
import { Provider } from 'react-redux';
// store
import { store } from './store';

interface ReduxProviderInterface {
  children: React.ReactNode;
  reduxStore: typeof store;
}

function ReduxProvider({ children, reduxStore }: ReduxProviderInterface): React.ReactElement {
  return <Provider store={reduxStore}>{children}</Provider>;
}

export default ReduxProvider;
