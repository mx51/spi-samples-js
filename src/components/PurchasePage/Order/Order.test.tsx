import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import * as redux from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';

import '@testing-library/jest-dom/extend-expect';
import Order from '.';
import ReduxProvider from '../../../redux/ReduxProvider';
import { store } from '../../../redux/store';

describe('Test <Order />', () => {
  test('snapshot for Order component', () => {
    const { container } = render(
      <ReduxProvider reduxStore={store}>
        <Router>
          <Order disablePayNow={false} isSubtotalEditable={false} bottomButton="payNow" />
        </Router>
      </ReduxProvider>
    );
    expect(container).toMatchSnapshot();
  });

  test('clearAll should clear the products', () => {
    const spy = jest.spyOn(redux, 'useDispatch');
    const dispatch = jest.fn();
    spy.mockReturnValue(dispatch);

    const { getByText } = render(
      <ReduxProvider reduxStore={store}>
        <Router>
          <Order disablePayNow={false} isSubtotalEditable={false} bottomButton="payNow" />
        </Router>
      </ReduxProvider>
    );

    fireEvent.click(getByText(/clear all/i));
    expect(dispatch.mock.calls.length).toBe(1);
    expect(dispatch.mock.calls[0][0]).toEqual({ payload: undefined, type: 'product/clearAllProducts' });
  });

  test('When click on Add button opens keypad', () => {
    const spy = jest.spyOn(redux, 'useDispatch');
    const dispatch = jest.fn();
    spy.mockReturnValue(dispatch);

    const { getByLabelText, getAllByLabelText, getByText } = render(
      <ReduxProvider reduxStore={store}>
        <Router>
          <Order disablePayNow={false} isSubtotalEditable={false} bottomButton="payNow" />
        </Router>
      </ReduxProvider>
    );

    fireEvent.click(getAllByLabelText(/add amount/i)[2]);
    expect(getByText(/enter tip amount/i)).toBeVisible();

    fireEvent.click(getByText(/5/i));
    fireEvent.click(getByText(/OK/i));

    expect(dispatch.mock.calls.length).toBe(1);
    expect(dispatch.mock.calls[0][0]).toEqual({ payload: 500, type: 'product/addTipAmount' });
    dispatch.mockReset();

    fireEvent.click(getAllByLabelText(/add amount/i)[0]);
    expect(getByText(/enter surcharge amount/i)).toBeVisible();

    fireEvent.click(getAllByLabelText(/add amount/i)[1]);
    expect(getByText(/enter cashout amount/i)).toBeVisible();

    fireEvent.click(getByLabelText(/close button/i));
    expect(dispatch.mock.calls.length).toBe(0);
  });
});
