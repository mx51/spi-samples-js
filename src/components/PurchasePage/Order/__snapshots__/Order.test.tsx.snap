// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <Order /> snapshot for Order component 1`] = `
<div>
  <div
    class="MuiPaper-root MuiBox-root MuiBox-root-17 makeStyles-root-1 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      class="MuiBox-root MuiBox-root-18 makeStyles-heading-2"
    >
      <h1
        class="MuiTypography-root MuiTypography-h6"
      >
        Order
      </h1>
      <div
        class="MuiBox-root MuiBox-root-19"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-clear-3"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Clear all
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
    <hr
      class="MuiDivider-root"
    />
    <div
      class="MuiBox-root MuiBox-root-20 makeStyles-orderList-12"
    >
      <ul
        class="MuiList-root MuiList-padding"
      />
    </div>
    <hr
      class="MuiDivider-root"
    />
    <ul
      class="MuiList-root MuiList-padding"
    >
      <li
        class="MuiListItem-root MuiListItem-gutters"
      >
        <div
          class="MuiListItemText-root"
        >
          <span
            class="MuiTypography-root MuiListItemText-primary makeStyles-subTotal-21 MuiTypography-body1 MuiTypography-displayBlock"
          >
            Subtotal
          </span>
        </div>
        <p
          class="MuiTypography-root makeStyles-price-22 MuiTypography-body1"
        >
          $0.00
        </p>
      </li>
      <li
        class="MuiListItem-root MuiListItem-gutters"
      >
        <div
          class="MuiListItemText-root"
        >
          <span
            class="MuiTypography-root MuiListItemText-primary makeStyles-additionalChargesEnable-23 MuiTypography-body1 MuiTypography-displayBlock"
          >
            Surcharge
          </span>
        </div>
        <span
          aria-disabled="false"
          aria-label="Add Amount"
          class="MuiButtonBase-root MuiIconButton-root makeStyles-addAmountBtn-27"
          role="button"
          tabindex="0"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M13 7h-2v4H7v2h4v4h2v-4h4v-2h-4V7zm-1-5C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </span>
      </li>
      <li
        class="MuiListItem-root MuiListItem-gutters"
      >
        <div
          class="MuiListItemText-root"
        >
          <span
            class="MuiTypography-root MuiListItemText-primary makeStyles-additionalChargesEnable-23 MuiTypography-body1 MuiTypography-displayBlock"
          >
            Cashout
          </span>
        </div>
        <span
          aria-disabled="false"
          aria-label="Add Amount"
          class="MuiButtonBase-root MuiIconButton-root makeStyles-addAmountBtn-27"
          role="button"
          tabindex="0"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M13 7h-2v4H7v2h4v4h2v-4h4v-2h-4V7zm-1-5C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </span>
      </li>
      <li
        class="MuiListItem-root MuiListItem-gutters"
      >
        <div
          class="MuiListItemText-root"
        >
          <span
            class="MuiTypography-root MuiListItemText-primary makeStyles-additionalChargesEnable-23 MuiTypography-body1 MuiTypography-displayBlock"
          >
            Tip
          </span>
        </div>
        <span
          aria-disabled="false"
          aria-label="Add Amount"
          class="MuiButtonBase-root MuiIconButton-root makeStyles-addAmountBtn-27"
          role="button"
          tabindex="0"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M13 7h-2v4H7v2h4v4h2v-4h4v-2h-4V7zm-1-5C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </span>
      </li>
      <hr
        class="MuiDivider-root makeStyles-divider-16 MuiDivider-middle"
      />
      <label
        class="MuiFormControlLabel-root makeStyles-checkbox-13"
      >
        <span
          aria-disabled="false"
          class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-28 MuiCheckbox-root MuiCheckbox-colorPrimary makeStyles-checkboxBase-15 MuiIconButton-colorPrimary"
        >
          <span
            class="MuiIconButton-label"
          >
            <input
              class="PrivateSwitchBase-input-31"
              data-indeterminate="false"
              name="Prompt for Cashout"
              type="checkbox"
              value=""
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </span>
        <span
          class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
        >
          <p
            class="MuiTypography-root makeStyles-checkboxLabel-14 MuiTypography-body1"
          >
            Prompt terminal for cashout
          </p>
        </span>
      </label>
      <li
        class="MuiListItem-root MuiListItem-gutters"
      >
        <div
          class="MuiListItemText-root"
        >
          <span
            class="MuiTypography-root MuiListItemText-primary makeStyles-total-5 MuiTypography-body1 MuiTypography-displayBlock"
          >
            Total
          </span>
        </div>
        <p
          class="MuiTypography-root makeStyles-totalPrice-6 MuiTypography-body1"
        >
          $0.00
        </p>
      </li>
    </ul>
    <a
      aria-disabled="true"
      class="MuiButtonBase-root MuiButton-root makeStyles-payNowBtn-7 MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeLarge MuiButton-sizeLarge Mui-disabled Mui-disabled"
      href="/paynow"
      role="button"
      tabindex="-1"
    >
      <span
        class="MuiButton-label makeStyles-actionBtnLabel-9"
      >
        Pay now
      </span>
    </a>
  </div>
</div>
`;
